google/cloud/storage/internal/curl_streambuf.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_CURL_STREAMBUF_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_CURL_STREAMBUF_H_
+
+#include "google/cloud/storage/internal/curl_upload_request.h"
+#include "google/cloud/storage/internal/object_streambuf.h"
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+/**
+ * Implement a wrapper for libcurl-based streaming uploads.
+ */
+class CurlStreambuf : public ObjectWriteStreambuf {
+ public:
+  explicit CurlStreambuf(CurlUploadRequest&& upload,
+                         std::size_t max_buffer_size = 64 * 1024);
```

Comment:
Any particular reason for 64K?

Reply:
Nope.  Bad idea actually, removed the default value.